src/windows/models/emoji_model.py

```diff
@@ -153,68 +158,45 @@ def update_model(self, clear=True):
 
-                # Append Media Type
-                col = QStandardItem("Type")
-                col.setData(type, Qt.DisplayRole)
-                col.setText(emoji_type)
-                col.setFlags(Qt.ItemIsSelectable | Qt.ItemIsEnabled | Qt.ItemIsUserCheckable | Qt.ItemIsDragEnabled)
-                row.append(col)
-
-                # Append Path
-                col = QStandardItem("Path")
-                col.setData(path, Qt.DisplayRole)
-                col.setText(path)
-                col.setFlags(Qt.ItemIsSelectable | Qt.ItemIsEnabled | Qt.ItemIsUserCheckable | Qt.ItemIsDragEnabled)
+                # Append filterable group
+                col = QStandardItem(emoji_type)
                 row.append(col)
 
                 # Append ROW to MODEL (if does not already exist in model)
-                if not path in self.model_paths:
+                if path not in self.model_paths:
                     self.model.appendRow(row)
                     self.model_paths[path] = path
 
-        # Get default emoji filter group
-        s = get_settings()
-        default_type = s.get('emoji_group_filter') or 'smileys-emotion'
-
-        # Loop through emoji groups, and populate emoji filter drop-down
-        get_app().window.emojiFilterGroup.clear()
-        get_app().window.emojiFilterGroup.addItem(_("Show All"), "Show All")
-        dropdown_index = 1
-        for emoji_type in sorted(emoji_groups.keys()):
-            get_app().window.emojiFilterGroup.addItem(_(emoji_type.capitalize()), emoji_type)
-            if emoji_type == default_type:
-                # Initialize emoji filter group to settings
-                get_app().window.emojiFilterGroup.setCurrentIndex(dropdown_index)
-            dropdown_index += 1
-
     def __init__(self, *args):
 
         # Create standard model
         self.app = get_app()
         self.model = EmojiStandardItemModel()
-        self.model.setColumnCount(4)
+        self.model.setColumnCount(2)
         self.model_paths = {}
+        self.emoji_groups = []
+
+        # Create proxy models (for grouping, sorting and filtering)
+        self.group_model = QSortFilterProxyModel()
+        self.group_model.setDynamicSortFilter(False)
+        self.group_model.setFilterCaseSensitivity(Qt.CaseInsensitive)
+        self.group_model.setSortCaseSensitivity(Qt.CaseSensitive)
+        self.group_model.setSourceModel(self.model)
+        self.group_model.setSortLocaleAware(True)
+        self.group_model.setFilterKeyColumn(1)
 
-        # Create proxy model (for sorting and filtering)
         self.proxy_model = QSortFilterProxyModel()
         self.proxy_model.setDynamicSortFilter(False)
         self.proxy_model.setFilterCaseSensitivity(Qt.CaseInsensitive)
         self.proxy_model.setSortCaseSensitivity(Qt.CaseSensitive)
-        self.proxy_model.setSourceModel(self.model)
+        self.proxy_model.setSourceModel(self.group_model)
         self.proxy_model.setSortLocaleAware(True)
```

Comment:
This was a major one. It felt to me like, since the selection list is based on categories, it made more sense to have _two_ proxy filters, one feeding into the other.

(That's [fully supported by the Qt framework](https://doc.qt.io/qt-5/model-view-programming.html#using-proxy-models):)
> Since proxy models inherit from QAbstractItemModel, they can be connected to any kind of view, and can be shared between views. They can also be used to process the information obtained from other proxy models in a pipeline arrangement.

By doing that, a selection from the dropdown list instructs `group_model` to select rows from a certain group, and then that list is passed to the `proxy_model` which can further filter it using the text field. I'll come back to that in comments on the view code.

Reply:
Very nice, never thought of that!